### PR TITLE
Re-Enable CA1416

### PR DESCRIPTION
**Pull request by PureWeen**
### Issues Fixed

Fixes #

### CODE REVIEW

File: src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs [hunk 1]
```diff
@@ -27,17 +28,36 @@ public static UIImage ScaleImage(this UIImage target, float maxWidth, float maxH
 
 		public static UIImage ScaleImage(this UIImage target, CGSize size, bool disposeOriginal = false)
 		{
-			UIGraphics.BeginImageContext(size);
-			target.Draw(new CGRect(CGPoint.Empty, size));
-			var image = UIGraphics.GetImageFromCurrentImageContext();
-			UIGraphics.EndImageContext();
-
-			if (disposeOriginal)
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```
Comment (jsuarezruiz) [inline src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:31]: Require this validation on more places (essentials etc):

`D:\a\_work\1\s\src\Essentials\src\Screenshot\Screenshot.ios.cs(89,4): error CA1416: This call site is reachable on: 'MacCatalyst' 13.1 and later. 'UIGraphics.EndImageContext()' is unsupported on: 'maccatalyst' 17.0 and later. `

File: src/Core/src/Handlers/SwipeItemMenuItem/SwipeItemMenuItemHandler.iOS.cs [hunk 2]
```diff
@@ -142,12 +142,32 @@ static UIImage MaxResizeSwipeItemIconImage(UIImage sourceImage, nfloat maxWidth,
...
+					return resultImage;
+				}
+
+				var format = new UIGraphicsImageRendererFormat
+				{
+					Opaque = false,
+					Scale = 0
+				};
+
+				using (var renderer = new UIGraphicsImageRenderer(new CGSize(width, height), format))
```
Comment (stephen-hawley) [inline src/Core/src/Handlers/SwipeItemMenuItem/SwipeItemMenuItemHandler.iOS.cs:161]: minor - there's no reason to use a block which you can just do:
```
using var renderer = new UIGraphicsImageRenderer(new CGSize(width, height), format);
```

File: src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs [hunk 3]
```diff
@@ -27,17 +28,36 @@ public static UIImage ScaleImage(this UIImage target, float maxWidth, float maxH
...
+					target.Dispose();
+				}
+
+				return image;
 			}
 
-			return image;
+			using (var renderer = new UIGraphicsImageRenderer(target.Size))
+			{
+				var resultImage = renderer.CreateImage((UIGraphicsImageRendererContext imageContext) =>
```
Comment (stephen-hawley) [inline src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:48]: same

File: src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs [hunk 4]
```diff
@@ -47,17 +67,35 @@ public static UIImage NormalizeOrientation(this UIImage target, bool disposeOrig
...
+				if (disposeOriginal)
+				{
+					target.Dispose();
+				}
+
+				return image;
 			}
 
-			return image;
+			using (var renderer = new UIGraphicsImageRenderer(target.Size))
```
Comment (stephen-hawley) [inline src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:85]: same

**Review by stephen-hawley — CHANGES_REQUESTED** [review]

**Review by stephen-hawley — DISMISSED** [review]

**Pull request closed without merging**